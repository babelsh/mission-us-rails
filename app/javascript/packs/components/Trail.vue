<template>
  <v-card>
    <v-card-media
      class="white--text"
      height="200px"
      :src="source"
    >
      <v-container fill-height fluid>
        <v-layout fill-height>
          <v-flex xs12 align-end flexbox>
            <span class="headline">{{ title }}</span>
          </v-flex>
        </v-layout>
      </v-container>
    </v-card-media>
    <v-card-actions>
      <v-btn icon>
        <v-icon :color="favoriteMarked">favorite</v-icon>
      </v-btn>
      <v-btn icon>
        <v-icon>bookmark</v-icon>
      </v-btn>
      <v-spacer></v-spacer>
      <v-btn color="info" v-if="isActive">
        Continuar
      </v-btn>
      <v-tooltip bottom v-else>
        <v-btn icon slot="activator">
          <v-icon>near_me</v-icon>
        </v-btn>
        <span>Iniciar trilha</span>
      </v-tooltip>
    </v-card-actions>
  </v-card>
</template>

<script>

export default {
  name: 'trail',
  props: {
    isActive: {
      type: Boolean,
      default: false,
    },
    isFavorite: {
      type: Boolean,
      default: false,
    },
    title: String,
    source: String,
  },
  computed: {
    favoriteMarked() {
      return this.isFavorite ? 'red' : 'default';
    }
  }
};

</script>
