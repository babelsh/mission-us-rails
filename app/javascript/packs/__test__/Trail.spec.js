import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import Trail from '../components/Trail'


describe('Trail.vue', () => {
  let localVue

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(Vuetify)
  })

  it('renders the Trail component', () => {

    const wrapper = mount(Trail, { title: 'test', localVue });

    expect(wrapper.html()).toMatchSnapshot()

  })
})
