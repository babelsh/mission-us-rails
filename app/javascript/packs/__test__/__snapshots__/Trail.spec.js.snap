// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trail.vue renders the Trail component 1`] = `"<div class=\\"v-card\\"><div class=\\"white--text v-card__media\\" style=\\"height: 200px;\\"><div class=\\"v-card__media__content\\"><div class=\\"container fill-height fluid\\"><div class=\\"layout fill-height\\"><div class=\\"flex xs12 align-end flexbox\\"><span class=\\"headline\\"></span></div></div></div></div></div> <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--icon\\"><div class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon default--text material-icons\\">favorite</i></div></button> <button type=\\"button\\" class=\\"v-btn v-btn--icon\\"><div class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon material-icons\\">bookmark</i></div></button> <div class=\\"spacer\\"></div> <span class=\\"v-tooltip v-tooltip--bottom\\"><div class=\\"v-tooltip__content\\" style=\\"left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;\\"> <span>Iniciar trilha</span></div><span><button type=\\"button\\" class=\\"v-btn v-btn--icon\\"><div class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon material-icons\\">near_me</i></div></button></span></span></div></div>"`;
